browser/ui/views/tabs/brave_compound_tab_container.cc

```diff
@@ -36,14 +36,24 @@ void BraveCompoundTabContainer::SetAvailableWidthCallback(
   if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs))
     return;
 
-  if (!available_width_callback || tabs::features::ShouldShowVerticalTabs(
-                                       tab_slot_controller_->GetBrowser())) {
-    // Unlike Chromium Impl, Just pass the `available_width_callback` to
-    // child containers when it's vertical tabs or we're trying to clear
-    // the callback.
-    pinned_tab_container_->SetAvailableWidthCallback(available_width_callback_);
-    unpinned_tab_container_->SetAvailableWidthCallback(
-        available_width_callback_);
+  if (tabs::features::ShouldShowVerticalTabs(
+          tab_slot_controller_->GetBrowser())) {
+    if (available_width_callback) {
+      pinned_tab_container_->SetAvailableWidthCallback(
+          base::BindRepeating(&views::View::width, base::Unretained(this)));
+      unpinned_tab_container_->SetAvailableWidthCallback(
+          base::BindRepeating(&views::View::width, base::Unretained(this)));
+    } else {
+      // we're trying to clear the callback.
+      pinned_tab_container_->SetAvailableWidthCallback(
+          available_width_callback_);
+      unpinned_tab_container_->SetAvailableWidthCallback(
+          available_width_callback_);
+    }
+  } else {
+    // Upstream's compound tab container doesn't use this.
+    pinned_tab_container_->SetAvailableWidthCallback(base::NullCallback());
+    unpinned_tab_container_->SetAvailableWidthCallback(base::NullCallback());
   }
```

Comment:
nit: How about simply this?
```suggestion
  if (tabs::features::ShouldShowVerticalTabs(
          tab_slot_controller_->GetBrowser()) && available_width_callback) {
    pinned_tab_container_->SetAvailableWidthCallback(
        base::BindRepeating(&views::View::width, base::Unretained(this)));
    unpinned_tab_container_->SetAvailableWidthCallback(
        base::BindRepeating(&views::View::width, base::Unretained(this)));
    return;
  }
  
  // Upstream's compound tab container doesn't use this.
  pinned_tab_container_->SetAvailableWidthCallback(base::NullCallback());
  unpinned_tab_container_->SetAvailableWidthCallback(base::NullCallback());
```

Reply:
That's smart!